### ISSUE TITLE
Mk[D]Tuple2._1 works where [d]fst doesn't

### ISSUE THREAD

**Issue by ckeller** [FStarLang/FStar#203]
I can type projections for dependent pairs this way:
```
val dfst : 'a:Type -> 'b:('a -> Type) -> DTuple2 'a 'b -> Tot 'a
let dfst t = MkDTuple2._1 t

val dsnd : 'a:Type -> 'b:('a -> Type) -> t:(DTuple2 'a 'b) -> Tot ('b (MkDTuple2._1 t))
let dsnd t = MkDTuple2._2 t
```
but this (δ-equivalent) way:
```
val dfst : 'a:Type -> 'b:('a -> Type) -> DTuple2 'a 'b -> Tot 'a
let dfst t = MkDTuple2._1 t

val dsnd : 'a:Type -> 'b:('a -> Type) -> t:(DTuple2 'a 'b) -> Tot ('b (dfst t))
let dsnd t = MkDTuple2._2 t
```
raises the error `Subtyping check failed; expected type ('b (dfst t)); got type (MkDTuple2.'b t (MkDTuple2._1 t))`.

**Comment by catalin-hritcu** [comment]
This appears not just with dependent pairs, but also with normal pairs. It seems the same problem as #173. This one has a much smaller test case though, so will close #173.

**Comment by catalin-hritcu** [comment]
This seems fixed now, but #173 is not (will reopen that one).
